skellytracker/trackers/yolo_mediapipe_combo_tracker/yolo_mediapipe_combo_tracker.py

```diff
@@ -43,23 +46,40 @@ def __init__(
 
         pytorch_model = yolo_object_model_dictionary[model_size]
         self.model = YOLO(pytorch_model)
+        self.bounding_box_buffer_percentage = bounding_box_buffer_percentage
 
     def process_image(self, image: np.ndarray, **kwargs) -> Dict[str, TrackedObject]:
+        
         yolo_results = self.model(image, classes=0, max_det=1, verbose=False)
-
         box_xyxy = np.asarray(yolo_results[0].boxes.xyxy).flatten()
 
+        width_buffer = image.shape[1] * (self.bounding_box_buffer_percentage / 100.0)
+        height_buffer = image.shape[0] * (self.bounding_box_buffer_percentage / 100.0)
+
         if box_xyxy.size > 0:
             box_left, box_top, box_right, box_bottom = box_xyxy
+
+            # Apply buffer, but set to original picture dimension if it goes out of bounds
+            box_left = max(int(box_left - width_buffer), 0)
+            box_top = max(int(box_top - height_buffer), 0)
+            box_right = min(int(box_right + width_buffer), image.shape[1])
+            box_bottom = min(int(box_bottom + height_buffer), image.shape[0])
+
             cropped_image = image[
                 int(box_top) : int(box_bottom),
                 int(box_left) : int(box_right),
             ]
+
+            buffered_yolo_results = copy.deepcopy(yolo_results)
+            buffered_yolo_results[0].boxes.xyxy[0] = torch.tensor([box_left, box_top, box_right, box_bottom])
+
         else:
             # eventually we should not even run mediapipe if no bbox is found
             box_left, box_top, box_right, box_bottom = 0, 0, 0, 0
             cropped_image = image
```

Comment:
This could be the subject of a new PR, but while we're working on this, I'll point it out.

I wrote this this way probably because it was easy, and without much testing I didn't want to trust YOLO as the only source of truth. But we could probably remove more "ghost skeletons" and reduce some processing overhead by just trusting YOLO and skipping mediapipe on these frames. Do you have any thoughts on this?

Reply:
Honestly, yeah I think I'd agree with this! In all my YOLO runs I don't think I've seen anything to suggest that there were times when mediapipe accurately detected a human that YOLO didn't. YOLO seems to be pretty solid at finding humans.

Reply:
Sweet, I'll work on that then.